bushnell.cc

```diff
@@ -168,20 +169,18 @@ rd_deinit(void)
 }
 
 static void
-wr_init(const char* fname)
+wr_init(const QString& fname)
 {
-  char* dot, *slash;
   static char valid_chars [] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ 0123456789"
                                ".-/\\~@#$%^&*()_+=<>"
                                "abcdefghijklmnopqrstuvwxyz";
 
-  ofname = xstrdup(fname);
-
   // If user provided an extension in the pathname, whack it.
-  dot = strrchr(ofname, '.');
-  slash = strrchr(ofname, GB_PATHSEP);
-  if (dot > slash) {
-    *dot = 0;
+  ofname = fname;
+  int suffix_len = QFileInfo(fname).suffix().length();
+  if (suffix_len > 0) {
+    /* drop the suffix and the period */
+    ofname.chop(suffix_len + 1);
```

Comment:
Is this ofname = QFileInfo(fname).basename() ? Maybe a little more if we support directory names here.
Given where Bushnell is in the GPS industry, I'm not sure I care much about doing this well.